src/common/snippets/src/shape_inference/shape_inference.cpp

```diff
@@ -88,6 +84,8 @@ std::shared_ptr<IShapeInferSnippets> make_shape_inference(const std::shared_ptr<
                ov::is_type<ov::op::util::BinaryElementwiseComparison>(op) ||
                ov::is_type<ov::op::util::BinaryElementwiseLogical>(op)) {
         return std::make_shared<NumpyBroadcastShapeInfer>();
+    } else if (ov::is_type<ov::snippets::op::ReduceBase>(op)) {
+        return std::make_shared<ReduceShapeInfer>(op);
```

Comment:
I know I'm being meticulous, but it would be more efficient to add this shape_infer for `ReduceMax` and `ReduceSum` through the ` SHAPE_INFER_PREDEFINED` macros. You see, lookup complexity in unordered_map is O(1), so additional elements in the map are free. However and `else if` block requires an extra type_info check in runtime.

Reply:
Thanks for the suggestion, changed